arbnode/transaction_streamer.go

```diff
@@ -1775,25 +1736,53 @@ func (s *TransactionStreamer) espressoSwitch(ctx context.Context, ignored struct
 		return retryRate
 	}
 	espressoMerkleProofEphemeralErrorHandler.Reset()
+	return s.espressoTxnsPollingInterval
+}
 
+/**
+ * Submits the transactions to espresso if the escape hatch is not enabled
+ */
+func (s *TransactionStreamer) submitTransactionsToEspresso(ctx context.Context, ignored struct{}) time.Duration {
+	retryRate := s.espressoTxnsPollingInterval * 50
 	shouldSubmit := s.shouldSubmitEspressoTransaction()
+	// Only submit the transaction if escape hatch is not enabled
 	if shouldSubmit {
-		s.submitEspressoTransactions(ctx)
-		return s.espressoTxnsPollingInterval
+		err := s.submitEspressoTransactions(ctx)
+		if err != nil {
+			log.Error("failed to submit espresso transactions", "err", err)
+			return retryRate
+		}
 	}
-
 	return s.espressoTxnsPollingInterval
 }
 
 func (s *TransactionStreamer) shouldSubmitEspressoTransaction() bool {
+	if s.espressoClient == nil && s.lightClientReader == nil {
+		return false
+	}
 	return !s.EscapeHatchEnabled
 }
 
 func (s *TransactionStreamer) Start(ctxIn context.Context) error {
 	s.StopWaiter.Start(ctxIn, s)
 
 	if s.lightClientReader != nil && s.espressoClient != nil {
-		err := stopwaiter.CallIterativelyWith[struct{}](&s.StopWaiterSafe, s.espressoSwitch, s.newSovereignTxNotifier)
+		err := stopwaiter.CallIterativelyWith[struct{}](&s.StopWaiterSafe, s.pollSubmittedTransactionForFinality, s.newSovereignTxNotifier)
+		if err != nil {
+			return err
+		}
+		err = stopwaiter.CallIterativelyWith[struct{}](&s.StopWaiterSafe, s.submitTransactionsToEspresso, s.newSovereignTxNotifier)
+		if err != nil {
+			return err
+		}
+
+		err = stopwaiter.CallIterativelyWith[struct{}](&s.StopWaiterSafe, func(ctx context.Context, ignored struct{}) time.Duration {
+			shouldSubmit := s.shouldSubmitEspressoTransaction()
+			if shouldSubmit {
+				s.submitEspressoTransactions(ctx)
+			}
+			return s.espressoTxnsPollingInterval
+		}, s.newSovereignTxNotifier)
```

Comment:
I'm confused by this. We have two separate async tasks. One is looping `submitTransactionsToEspresso` and one is looping `submitEspressoTransactions`. Aside from the naming being confusing there, it looks like `submitTransactionsToEspresso` also just calls `submitEspressoTransactions`, so why do we need both of these?

Reply:
You are right. merge mistakes